### DISCUSSION TITLE
SimpleObject with PrismaField

### DISCUSSION THREAD

**Discussion by enri90** [hayes/pothos#1297] [Q&A]
Hi everyone I ask for help to solve this problem with prisma if its prismaField does not work 
if I import prismaObject with field yes why?

```typescript
t.field({
    type: ListPayload,
    args: {     
      orderBy: t.arg({ type: MoviesOrderByInput, required: false }),
      skip: t.arg.int({ required: false, defaultValue: 0 }),
      take: t.arg.int({ required: false, defaultValue: 20 }),
    },
    resolve: async (query, args, ctx) => {
      const [total, list] = await ctx.prisma.$transaction(
      return {
          total,
          list
      }
``` 

this is not working before it worked in previous version
```typescript
export const ListPayload = builder.simpleObject('ListPayload', {
  description: '',
  fields: (t) => ({
    total: t.int(),
    list: t.prismaField({
      type: ['Page'],
      nullable: true,
      resolve: (query, parent, args, ctx, info) => []
    })
  })
})

``` 

this works
```typescript
import Page from ''

export const ListPayload = builder.simpleObject('ListPayload', {
  description: '',
  fields: (t) => ({
    total: t.int(),
    list: t.field({
      type: [Page],
    })
  })
})

```

**Comment by hayes** [comment]
fields for simple objects can't have resolvers.  The primary purpose for simple objects is to infer the data requirements for the object type from the fields.

If your data is being provided from the parent field you don't need `prismaField`:

```ts
const Page = builder.prismaObject({...})
export const ListPayload = builder.simpleObject('ListPayload', {
  description: '',
  fields: (t) => ({
    total: t.int(),
    list: t.field({
      type: [Page],
      nullable: true,
    })
  })
})
```

**Reply by hayes** [reply]
I see you already had this as your last example.  Is there a reason you don't want to do it this way?

**Reply by enri90** [reply]
the only thing I find inconvenient is having to import prismaObject when with prismaField the type field automatically reveals it.
also with variants of prismaObject and prismaNode

**Reply by hayes** [reply]
If you want to avoid the imports, you can add the types at the builder level:

```ts
const builder = new SchemaBuilder<{
  ...
  Objects: {
     Page: Prisma.Page
  }
}>({ ... });

export const ListPayload = builder.simpleObject('ListPayload', {
  description: '',
  fields: (t) => ({
    total: t.int(),
    list: t.field({
      type: ['Page'],
      nullable: true,
    })
  })
})
```

**Reply by enri90** [reply]
and as prisma variants how do I set it?
Thank you

**Reply by hayes** [reply]
the keys are the GraphQL type names, not the prisma model names, so it would be whatever the variants name is
